src/edit.c

```diff
@@ -929,6 +929,8 @@ edit(
 	    break;
 
 	case Ctrl_R:	// insert the contents of a register
+	    if (ctrl_x_mode_register() && !ins_compl_active())
+		goto docomplete;
```

Comment:
When is this condition ever true?

Reply:
`<C-X>` then `<C-R>` edit invoke  ins_compl_prep in 707 then ins_compl_prep invoke set_ctrl_x_mode which set `ctrl_x_mode` to reigsert mode , so when run in there the `ctrl_x_mode` has been set

Reply:
Oh, right.